tests/foreman/ui/test_discoveredhost.py

```diff
@@ -98,16 +100,135 @@ def test_positive_provision_pxe_host(
     # Teardown
     request.addfinalizer(lambda: sat.provisioning_cleanup(host_name))
 
-    with session:
+    with sat.ui_session() as session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_location.name)
         session.discoveredhosts.provision(
             discovered_host_name,
             provisioning_hostgroup.name,
             module_org.name,
             module_location.name,
         )
-        values = session.host.get_details(host_name)
-        assert values['properties']['properties_table']['Status'] == 'OK'
+        # Wait for provisioning to complete and report status back to Satellite
+        pending_status = 'N/A' if is_open('BZ:2258024') else 'Pending installation'
+        wait_for(
+            lambda: session.host_new.get_host_statuses(host_name)['Build']['Status']
+            != pending_status,
+            timeout=1800,
+            delay=30,
+            fail_func=session.browser.refresh,
+            silent_failure=True,
+            handle_exception=True,
+        )
+        values = session.host_new.get_host_statuses(host_name)
+        assert values['Build']['Status'] == 'Installed'
+
+        # Verify entry from discovered host is auto removed
+        assert not session.discoveredhosts.search(f'name = {discovered_host_name}')
+
+
+@pytest.mark.tier3
+@pytest.mark.upgrade
+@pytest.mark.on_premises_provisioning
+@pytest.mark.parametrize('module_provisioning_sat', ['discovery'], indirect=True)
```

Comment:
Since we are doing full provisioning you can remove this as we would need the content repos as well
```suggestion
```

Reply:
Updated